### PR TITLE
Invoice Sync Controller Overhaul

### PR DESCRIPTION
**Pull request by JRising-Xero**
- Removed redundant code
- Converted variable names to use correct c# conventions
- Extracted repeated code to helper methods
- Added comments and summaries 
- Isolated methods by HTTP type using regions
- Removed unnecessary dependency injection

### CODE REVIEW

File: XeroNetStandardApp/Utilities/TokenUtilities.cs [hunk 1]
```diff
@@ -20,6 +24,23 @@ public struct State
       this.state = state;
     }
   }
+
+
+  public async static Task<XeroOAuth2Token> GetXeroOAuth2Token(XeroConfiguration xeroConfig)
```
Comment (JRising-Xero) [inline XeroNetStandardApp/Utilities/TokenUtilities.cs:29]: It will be good to use this static method across all controllers and set the existing GetStoredToken() to a private static method. 
Will do in a different PR to keep this PR focused on the InvocieSyncController

File: XeroNetStandardApp/Utilities/TokenUtilities.cs [hunk 2]
```diff
@@ -109,4 +130,23 @@ public static string GetCurrentState()
 
     return state;
   }
+
+    public static string GetXeroTenantId(XeroOAuth2Token xeroToken)
```
Comment (JRising-Xero) [inline XeroNetStandardApp/Utilities/TokenUtilities.cs:134]: Same as comment above

**Review by rlayhbrian — APPROVED** [review]
All good from my side, awesome work! :D
